zerver/actions/submessage.py

```diff
@@ -48,6 +51,29 @@ def do_add_submessage(
     )
     submessage.save()
 
+    # Determine and set the visibility_policy depending on 'automatically_follow_topics_policy'
+    # and 'automatically_unmute_topics_policy'.
+    if submessage.message.is_stream_message():
+        stream_id = submessage.message.recipient.type_id
```

Comment:
This function doesn't check the `is_bot` case that the reactions code path does; should it? I think maybe bots can't use submessages, but it might be more future-proof to to the exact check we have for reactions here.

Reply:
Used `set_visibility_policy_possible` as explained in the above comment.